libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -182,7 +182,7 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
                 }
             }
 
-            var activeDialogState = dc.ActiveDialog.State as Dictionary<string, object>;
+            var activeDialogState = dc.ActiveDialog.State;
```

Comment:
these changes in Adaptive mean that we can use types None in adaptive dialogs - is that correct?

Reply:
and can we separate the issues and take this fixes to adaptive independently of the changes to classic

Reply:
yes it does, and there is no reason to take the changes seperately.

Reply:
That's good news. The suggestion to separate is the changes is that we want to be very explicit about the change in behavior on existing code when things are switched. Many people simply won't care. It would be fine to make the extension function on IDictionary internal.

For those users who are on old Dialogs _and_ care about this I don't think the extension function is that helpful. If people who both care and struggle with this need help in this regard they can grab the relevant couple of lines of code if they need to. The more important issue in all this is that we document the implications of this change. When we have done that the helper (the extension) is of little value in the product, if not counterproductive.

The high order bit here is that the new dialog code works find with None.